gplugins/common/utils/get_component_with_net_layers.py

```diff
@@ -67,17 +66,23 @@ def get_component_with_net_layers(
     for i, portname in enumerate(port_names):
         port = component.ports[portname]
         # Get original port layer polygons, and modify a new component without that layer
-        polygons = net_component.extract(layers=[port.layer]).get_polygons()
-        net_component = net_component.remove_layers(layers=[port.layer])
+        polygons = (
+            net_component.extract(layers=(port.layer,))
+            .get_polygons()
+            .get(port.layer, [])
+        )
+        net_component = net_component.remove_layers(layers=(port.layer,))
         for polygon in polygons:
             # If polygon belongs to port, create a unique new layer, and add the polygon to it
-
-            if gdstk.inside(
-                [port.center],
-                gdstk.offset(gdstk.Polygon(polygon), gf.get_active_pdk().grid_size),
-            )[0]:
+            if polygon.sized(0.005).inside(port.center):
```

Comment:
do we still have a grid size we can use here? Is hardcoding this value safe?

Reply:
Fixed it to use the dbu